### PR TITLE
Fix arbitrage swap double entry bug

### PR DESCRIPTION
**Pull request by gheise**
## Resolution to Issue #145 
As highlighted in the issue, there was a bug in getting arbitrages where, in certain cases, a start/end pair with equal entries would be passed as a valid. This causes the arbitrage swaps list to contain a double entry, which breaks when writing to the db due to a non-unique primary key. The error occurred every couple hundred blocks. A check can be introduced to remove this from the valid start/end pairs.

## TLDR
- Introduce a check for valid arbitrage start/end pairs where the start/end swaps cannot be the same swap

### CODE REVIEW

File: mev_inspect/arbitrages.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ def _get_all_start_end_swaps(swaps: List[Swap]) -> List[Tuple[Swap, Swap]]:
                 == potential_end_swap.token_out_address
                 and potential_start_swap.from_address == potential_end_swap.to_address
                 and not potential_start_swap.from_address in pool_addrs
+                and potential_start_swap != potential_end_swap
```
Comment (lukevs) [inline mev_inspect/arbitrages.py:98]: rather than comparing the objects directly which can sometimes be funky, I think it'd be slightly better to explicitly say what you mean which is "skip the swap I have as the start swap"

something like:
```
for index, potential_start_swap in enumerate(swaps):
    other_swaps = swaps[:index] + swaps[index+1:]
    for potential_end_swap in other_swaps:
        ...
```

Comment (gheise) [reply mev_inspect/arbitrages.py]: Thanks for the feedback, this was implemented

**Review by lukevs — APPROVED** [review]
